zebra-scan/src/service.rs

```diff
@@ -130,8 +131,33 @@ impl Service<Request> for ScanService {
                 .boxed();
             }
 
-            Request::Results(_key_hashes) => {
-                // TODO: read results from db
+            Request::Results(keys) => {
+                let db = self.db.clone();
+
+                return async move {
+                    if keys.len() > MAX_REQUEST_KEYS {
+                        return Err(format!(
+                            "maximum number of keys per request is {MAX_REQUEST_KEYS}"
+                        )
+                        .into());
+                    }
```

Comment:
Will be removed after https://github.com/ZcashFoundation/zebra/pull/8219 gets merged.

Reply:
Removed as part of https://github.com/ZcashFoundation/zebra/pull/8224/commits/b6d7003f569d1198ffb7d1d2630499627d7883a3